# Задача 1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

$\mu = 17,   
\alpha = 0.025,  
n = 100 , 
S = 17.5, 
\sigma = 4$

Нулевая гипотеза $H0$ - подшипники имеют средний диаметр 17мм, альтернативная гипотеза $H1$ - средний диаметр подшипников не равен 17мм

$t = \frac{\bar{X}-m} {s_x/ \sqrt n} $ 

In [50]:
t =(17.5 - 17) / (2 / (pow(100, .5)))
t

2.5

Высчитаем степень свободы

In [9]:
df = 100 - 1 
df

99

Расчитаем p-value для нормального распределения

In [11]:
from scipy.stats import norm, t
import numpy as np

p = norm.ppf(1 - 0.025)
p

1.959963984540054

## Ответ
Так как t-критерий больше чем p-value мы можем отвергнуть нулевую гипотезу

# Задача 2

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

$\bar{X} = 80, n = 256, \sigma = 16 $

Найдем доверительный интервал с ожидаемой надежностью 0.95 по формуле 

$\hat{x} - t_\gamma\frac{\sigma}{\sqrt{n}} < a < \hat{x} + t_\gamma\frac{\sigma}{\sqrt{n}}$

Где $t_y$ определяется по таблице значений функции Лапласа из равенства

$Ф(t_\gamma) = \frac{\gamma}{2}$

In [23]:
y = 0.95
F = y/2
F

0.475

По таблице Лапласса получаем $t_\gamma = 1,96$

Подставляем значения в формулу

In [25]:
80 - 1.96 * (16 / 256 ** .5)

78.04

In [26]:
80 + 1.96 * (16 / 256 ** .5)

81.96

## Ответ

$78.04 < a < 81.96$

# Задача 3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [35]:
import numpy as np
a = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

$\bar{X} = 200$

$H_0$ - продавец прав, и средний вес пачки равен 200г <br>
$H_1$ - средний вес пачки отличается от указанного продавцом

Считаем среднее $\overline{x}$ и несмещенную дисперсию $\sigma_x$

In [37]:
x = a.mean()
s = np.std(a, ddof=1)
x, s

(198.5, 4.453463071962462)

Посчитаем значение t-статистики

$t = \frac{\bar{X}-m} {s_x/ \sqrt n} $ 

In [48]:
t = (x - 200) / (s / (pow(10, .5)))
t

-1.0651074037450896

Считаем квартили уровня значимости с права и лева по $0.5%$

In [44]:
from scipy.stats import norm, t
t1 = t.ppf(0.005, df=9)
t2 = t.ppf(0.995, df=9)
t1, t2

(-3.24983554401537, 3.2498355440153697)

## Ответ

Так как $t$-статистика попадает в интервал, мы может отвергнуть $H_0$ гипотезу